### PR TITLE
get checkpoint from specified consumer group

### LINKED ISSUE #210: $Default consumer group used to retrieve the checkpoint regardless of the one set in the opts
### Expected Behavior
If I set a consumer group on the Receive(), I expect it to be used to invoke the persister.

### Actual Behavior
Instead the persister, at the very first time it's invoked, it receives "@Default"

It seems a problem with the order in which the opts are applied to the receiver struct.

### Environment
Environment
OS: alpine:3.10.6
Go version: 1.16.2
Version of Library: v3.3.7

### PR DESCRIPTION
**Pull request by catalinaperalta**
Fixes #210

### CODE REVIEW

**Comment by catalinaperalta** [comment]
This PR may be unnecessary, if we add the changes in the PR about customizing the intial checkpoint.

**Comment by catalinaperalta** [comment]
So while the other PR and this one are related. This issue is also about the receiver options being applied correctly to set up the receiver, so I think we should still make this change.

File: receiver.go [hunk 1]
```diff
@@ -158,7 +152,16 @@ func (h *Hub) newReceiver(ctx context.Context, partitionID string, opts ...Recei
 		}
 	}
 
-	if err = receiver.storeLastReceivedCheckpoint(receiver.checkpoint); err != nil {
+	// update checkpoint if no checkpoint is specified and if old checkpoint is successfully read from e.g. file or memory
+	emptyCheckpoint := persist.Checkpoint{}
+	if receiver.checkpoint == emptyCheckpoint {
```
[the remark this answers is not in the document]
Comment (catalinaperalta) [reply receiver.go:157]: Yes it can, dont know why I did that. Will update this.

[the remark this answers is not in the document]
Comment (catalinaperalta) [reply receiver.go:157]: Actually, I'd have to do something like this: `if receiver.checkpoint == (persist.Checkpoint{})`, is that still good?

[the remark this answers is not in the document]
Comment (jhendrixMSFT) [reply receiver.go:157]: Oh yes, the `()` are required (I forgot about this).  It's explained under [here](https://golang.org/ref/spec#Composite_literals).

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**